components/timeline/src/Timeline.tsx

```diff
@@ -0,0 +1,51 @@
+import { FC, ReactNode, createElement as h } from 'react';
+import { StandardProps, classBuilder } from '@not-govuk/component-helpers';
+
+import '../assets/Timeline.scss';
+
+type TimelineItem = {
+  action: string
+  by?: string
+  date: string
+  description: ReactNode
+}
+
+export type TimelineProps = StandardProps & {
+  /** Items is a compulsory property of the Timeline component.
+  Items consists of an array of item objects, each consisting of an
+  action, the name of the person who performed the action (optional),
+  the date (and time) the action was taken and a description of the action.
+  */
+  items: TimelineItem[]
+};
+
+export const Timeline: FC<TimelineProps> = ({
+  children,
+  classBlock,
+  classModifiers,
+  className,
+  items,
+  ...attrs
+}) => {
+  const classes = classBuilder(
+    'hods-timeline',
+    classBlock,
+    classModifiers,
+    className
+  );
+
+  return (
+      <div className={classes()}>
+        {items.map(e => (
+          <div className={classes('item')}>
+            <h2 className={classes('title')}>{e.action}</h2>
```

Comment:
I think perhaps this should be an `h3` at least - so you can have:
```html
<h1>Page title</h1>
<h2>Section title / timeline title</h2>
<!-- timeline -->
  <!-- timeline-item -->
    <h3>Timeline item title</h3>
    <p>Timeline item description...</p>
  <!-- /timeline-item -->
<!-- /timeline -->
```

Reply:
Yeah that's a tricky one.

I'd almost think they should be `<div>` with `h2` styling on them.

Worth checking what GDS do on some of their components.

Reply:
Closest I can find is the [summary list](https://design-system.service.gov.uk/components/summary-list/).
It's a table, but they set `class="govuk-summary-list__key"` for header cells and `class="govuk-summary-list__value"` for other cells

Reply:
In Tabs they use an `h2` for their hidden heading: https://design-system.service.gov.uk/components/tabs/

Let's stick with `h2` for now.

Reply:
Will do.

Reply:
Tabs are usually a higher level component though (hence `h2`). If you look at this [section](https://design-system.service.gov.uk/styles/page-template/#options) you can see it is correctly marked as `h3` following the structure of the page. I'm worried that `h2` here is too high, any thoughts @karypun ?